packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/transformation/pureProtocol/V1_DSL_DataSpace_ProtocolHelper.ts

```diff
@@ -130,15 +136,59 @@ const V1_dataSpaceElementPointerModelSchema = createModelSchema(
   },
 );
 
-const V1_dataSpaceExecutableModelSchema = createModelSchema(
-  V1_DataSpaceExecutable,
+const V1_dataSpacePackageableElementExecutableModelSchema = createModelSchema(
+  V1_DataSpacePackageableElementExecutable,
   {
+    _type: usingConstantValueSchema(V1_DATA_SPACE_PACKAGEABLEMENT_EXECUTABLE),
     description: optional(primitive()),
+    title: primitive(),
     executable: usingModelSchema(V1_packageableElementPointerModelSchema),
+  },
+);
+
+const V1_dataSpaceTemplateExecutableModelSchema = createModelSchema(
+  V1_DataSpaceTemplateExecutable,
+  {
+    _type: usingConstantValueSchema(V1_DATA_SPACE_TEMPLATE_EXECUTABLE),
+    description: optional(primitive()),
     title: primitive(),
+    query: usingModelSchema(V1_rawLambdaModelSchema),
+    executionContextKey: optional(primitive()),
   },
 );
 
+const V1_serializeDataspaceExecutable = (
+  protocol: V1_DataSpaceExecutable,
+): PlainObject<V1_DataSpaceExecutable> => {
+  if (protocol instanceof V1_DataSpaceTemplateExecutable) {
+    return serialize(V1_dataSpaceTemplateExecutableModelSchema, protocol);
+  } else if (protocol instanceof V1_DataSpacePackageableElementExecutable) {
+    return serialize(
+      V1_dataSpacePackageableElementExecutableModelSchema,
+      protocol,
+    );
+  }
+  throw new UnsupportedOperationError(
+    `Can't serialize data space executable`,
+    protocol,
+  );
+};
+
+export const V1_deserializeDataspaceExecutable = (
+  json: PlainObject<V1_DataSpaceExecutable>,
+): V1_DataSpaceSupportInfo => {
+  switch (json._type) {
+    case V1_DATA_SPACE_TEMPLATE_EXECUTABLE:
+      return deserialize(V1_dataSpaceTemplateExecutableModelSchema, json);
+    case V1_DATA_SPACE_PACKAGEABLEMENT_EXECUTABLE:
+    default:
```

Comment:
How come with serialization we throw an error if the type is unexpected, but with deserialization we don't?

Reply:
This is to make sure this update of the metadata model is backward-compatible. There might be a Json where the _type is missing.
like
`      "executables": [
        {
          "title": "Executable 1",
          "description": "description",
          "executable": { "path": "model::MyExecutable" }
        },
        {
          "_type": "dataSpacePackageableElementExecutable",
          "title": "Executable 2",
          "description": "description",
          "executable": { "path": "model::MyExecutable" }
        },
        {
          "_type": "dataSpacePackageableElementExecutable",
          "title": "Executable 3",
          "description": "description",
          "executable": { "path": "model::MyExecutable" }
        }
      ],`